textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcherTest.java

```diff
@@ -12,56 +10,53 @@
 import edu.uci.ics.textdb.api.common.IPredicate;
 import edu.uci.ics.textdb.api.common.ITuple;
 import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.common.constants.LuceneConstants;
+import edu.uci.ics.textdb.common.constants.TestConstants;
 import edu.uci.ics.textdb.dataflow.common.RegexPredicate;
-import edu.uci.ics.textdb.dataflow.constants.LuceneConstants;
-import edu.uci.ics.textdb.dataflow.source.SampleDataStore;
 import edu.uci.ics.textdb.dataflow.source.ScanBasedSourceOperator;
-import edu.uci.ics.textdb.dataflow.source.TestConstants;
+import edu.uci.ics.textdb.dataflow.utils.TestUtils;
+import edu.uci.ics.textdb.storage.LuceneDataStore;
 
 /**
  * Created by chenli on 3/25/16.
  */
 public class RegexMatcherTest {
     
     private RegexMatcher regexMatcher;
-    private SampleDataStore dataStore;
+    private LuceneDataStore dataStore;
     
     @Before
     public void setUp() throws Exception{
-        dataStore = new SampleDataStore();
+        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR);
         dataStore.clearData();
-        dataStore.storeData();
+        dataStore.storeData(TestConstants.SAMPLE_SCHEMA, TestConstants.getSampleTuples());
```

Comment:
Yes it will be allowed. Since we are passing the schema and the tuples as the parameters to LuceneDataStore.storeData() method, they can create a new data store with another schema.